crates/nu-cli/src/commands/math/command.rs

```diff
@@ -0,0 +1,130 @@
+use crate::commands::WholeStreamCommand;
+use crate::prelude::*;
+use nu_errors::ShellError;
+use nu_protocol::{ReturnSuccess, Signature, UntaggedValue};
+
+pub struct Command;
+
+#[async_trait]
+impl WholeStreamCommand for Command {
+    fn name(&self) -> &str {
+        "math"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("math")
+    }
+
+    fn usage(&self) -> &str {
+        r#"Use mathematical functions (average, min, max) to aggregate list of numbers or tables
+        math average
+        math min
+        math max
+        "#
+    }
+
+    async fn run(
+        &self,
+        _args: CommandArgs,
+        registry: &CommandRegistry,
+    ) -> Result<OutputStream, ShellError> {
+        Ok(OutputStream::one(Ok(ReturnSuccess::Value(
+            UntaggedValue::string(crate::commands::help::get_help(&Command, &registry.clone()))
+                .into_value(Tag::unknown()),
+        ))))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::commands::math::{
+        average::average, max::maximum, min::minimum, utils::MathFunction,
+    };
+    use nu_plugin::test_helpers::value::{decimal, int};
+    use nu_protocol::Value;
+
+    #[test]
+    fn examples_work_as_expected() {
+        use crate::examples::test as test_examples;
+
+        test_examples(Command {})
+    }
+
+    #[test]
+    fn test_math_functions() {
+        struct TestCase {
+            description: &'static str,
+            values: Vec<Value>,
+            expected_err: Option<ShellError>,
+            // Order is: avg, min, max
+            expected_res: Vec<Result<Value, ShellError>>,
+        }
+        let tt: Vec<TestCase> = vec![
+            TestCase {
+                description: "Single value",
+                values: vec![int(10)],
+                expected_err: None,
+                expected_res: vec![Ok(decimal(10)), Ok(int(10)), Ok(int(10))],
+            },
+            TestCase {
+                description: "Multiple Values",
+                values: vec![int(10), int(30), int(20)],
+                expected_err: None,
+                expected_res: vec![Ok(decimal(20)), Ok(int(10)), Ok(int(30))],
+            },
+            TestCase {
+                description: "Mixed Values",
+                values: vec![int(10), decimal(26.5), decimal(26.5)],
+                expected_err: None,
+                expected_res: vec![Ok(decimal(21)), Ok(int(10)), Ok(decimal(26.5))],
+            },
+            TestCase {
+                description: "Negative Values",
+                values: vec![int(10), int(-11), int(-14)],
+                expected_err: None,
+                expected_res: vec![Ok(decimal(-5)), Ok(int(-14)), Ok(int(10))],
+            },
+            // TODO-Address once we figure out how to handle this. Maybe it's not an important use-case
```

Comment:
A test case that fails currently... I think it has something to do with improper coercion on `Primitive::min/max` calls